### ISSUE TITLE
Not easy enough for noobs

### ISSUE THREAD

**Issue by stevenirby** [StarshipEngineer/OpenVPN-Setup#4]
I think this is a good start, but could be a lot easier. Here are some suggestions:

* don't make the user apt-get install git; have the script do it if need be
* have just one command they copy and paste to download and run the script from github

Example:
```
curl https://raw.githubusercontent.com/StarshipEngineer/OpenVPN-Setup/master/openvpnsetup.sh | bash
```

* install with sane defaults if no args provided

Optional nice to have IMO:
 * download handy duckdns script for easy handling of dynamic IPs: http://duckdns.org/install.jsp

**Comment by StarshipEngineer** [comment]
Thanks for the feedback! I can touch on a couple of these ideas.

Installing git is only included in the instructions so that the script can be easily downloaded from Github via terminal or SSH, as I assumed the Pi user might not necessarily be using LXDE desktop. That might also make copy-pasting a command more difficult, but even so I think you're right and new users may prefer typing one longer command to multiple commands. I'll try it out and alter the Readme installation instructions to cover something like that, as well as just downloading a zip file of the repo to the Pi some other way.

As far as installing with defaults, I'll see what I can do. The internal and external IP addresses are a must though at the absolute minimum. However, the process could be simplified if I could figure out how to call scripts with arguments, rather than waiting for the user to input them, so I'll see what I can figure out (Linux newbie here). 

Dynamic DNS shouldn't be an issue with this setup, but I'll check out duckdns. I used Google DNS to try and get around that problem, and it seems to be working for my setup.

**Comment by stevenirby** [comment]
Yeah I guess a user might have to type in the command without copy and paste. I'm so use to ssh'ing into my pi from computer. You can always use shortened url:

```
curl http://bit.ly/1aHecXK | bash
```

In bash you can pass as many arguments you want to a script. In the bash you reference arguments like this:

arg1 - `$1`
arg2 - `$2`
arg3 - `$3`
All args - `"$@"`

You could also export something before running the script: `export foo=bar; ./script.sh`

**Comment by StarshipEngineer** [comment]
I considered positional arguments for a while, but there are a lot of optional fields that must be filled out or skipped through when generating the certificate authority and server certificate, and I think it would be more user-friendly to use another text file containing default values for these fields and have STDIN input these values into the script as it runs as follows:

sudo ./openvpnsetup.sh < defaultinput.txt

Where the defaultinput.txt file has the answer to one info field per line, or else a period or blank line if the user would rather not fill out any unnecessary fields.

However, every time I try this, the script has trouble generating a certificate request for the server for reasons I haven't figured out. I'm still trying to make this work, because it seems like the ideal solution for making things easier. Automating the few remaining manual input portions of the script would also allow me to build in a whiptail GUI, which I' like to do.

**Comment by StarshipEngineer** [comment]
Have had some success with this, and the script now requires less user input. Still having issues with using STDIN where the server certificate info is concerned.

**Comment by 0-kaladin** [comment]
I'm on this... stay tuned...
